--- functions.og	2020-06-16 06:44:57.888248386 -0400
+++ functions	2020-06-16 14:25:50.410150162 -0400
@@ -256,8 +256,8 @@
 	if [ -n "$dtb_name" ] ; then
 		echo "Using DTB: $dtb_name" >&2
 	fi
-	dtb_dir=$(dirname "$dtb_name")
-	dtb_name=$(basename "$dtb_name")
+	dtb_dir="qcom"
+	dtb_name="sdm850-lenovo-yoga-c630.dtb"
 }
 
 machine_uses_flash() {
@@ -662,7 +662,7 @@
 		;;
 	*)
 		dtb=
-		paths="/etc/flash-kernel/dtbs /usr/lib/linux-image-$kvers /lib/firmware/$kvers/device-tree/"
+		paths="/etc/flash-kernel/dtbs /usr/lib/linux-image-$kvers/qcom /lib/firmware/$kvers/device-tree/"
 		for path in $paths; do
 			dtb=$(find $path -name $dtb_name 2>/dev/null | head -n 1)
 			[ -z "$dtb" ] || break
@@ -690,7 +690,7 @@
 find_all_dtbs() {
 	# NOTE: The following assumes that "find" processes and outputs
 	# filenames in the same order as starting points ($paths) are specified
-	paths="/etc/flash-kernel/dtbs /usr/lib/linux-image-$kvers /lib/firmware/$kvers/device-tree/"
+	paths="/etc/flash-kernel/dtbs /usr/lib/linux-image-$kvers/qcom /lib/firmware/$kvers/device-tree/"
 	find $paths -name "*.dtb" 2>/dev/null | unique_filenames
 }
 
